packages/web3-eth-ens/src/ens.ts

```diff
@@ -195,15 +203,42 @@ export class ENS extends Web3Context<EthExecutionAPI> {
 		return this._resolver.getContenthash(ENSName);
 	}
 
+	/**
+	 * Checks if the current used network is synced and looks for ENS support there.
+	 * Throws an error if not.
+	 */
+	public async checkNetwork() {
+		const now = Date.now() / 1000;
+		if (!this._lastSyncCheck || now - this._lastSyncCheck > 3600) {
+			const block = await getBlock(this, 'latest');
+			const headAge = BigInt(now) - BigInt(block.timestamp);
+
+			if (headAge > 3600) {
+				throw new Error(`Network not synced; last block was ${headAge} seconds ago`);
+			}
+
+			this._lastSyncCheck = now;
+		}
+
+		if (this._detectedAddress) {
+			return this._detectedAddress;
+		}
+		const net = new Web3Net(this);
```

Comment:
Instead of creating new object here, good to explore: directly using rpc [wrapper](https://github.com/ChainSafe/web3.js/blob/2c557d6a8075e6f95f2fbda2668bd35191c81857/packages/web3-net/src/rpc_method_wrappers.ts#L7) internally , and passing context to that.